python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -215,6 +220,109 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info, fil
     return embedding
 
 
+class CategoricalDropout(layers.Layer):
+    """Custom Dropout class for categorical indices"""
+
+    def __init__(self, dropout_rate, seed=None, **kwargs):
+        """
+        Args:
+            dropout_rate: fraction of units to drop, i.e., set to OOV token 0
+            seed: random seed for sampling to mask/drop categorical labels
+
+        Note:
+        We define OOV index to be 0 for this function and when dropout is applied, it converts p% of the values to 0(which is the OOV index). This allows us to train a good average embedding for the OOV token.
```

Comment:
@lastmansleeping [EXTREMELY minor] Would be nice to put some PEP-8 length returns on this so one can see the whole note without scrolling :)